nltools/data/brain_data.py

```diff
@@ -1839,52 +1800,58 @@ def align(self, target, method='procrustes', n_features=None, axis=0,
 
         '''
 
-        source = self.copy()
-        common = target.copy()
-
-        target = check_brain_data(target)
-
         if method not in ['probabilistic_srm', 'deterministic_srm', 'procrustes']:
             raise ValueError("Method must be ['probabilistic_srm','deterministic_srm','procrustes']")
+        
+        source = self.copy()
+        data1 = self.data.copy()
 
-        data1 = source.data.T
-        data2 = target.data.T
+        if method == 'procrustes':
+            target = check_brain_data(target)
+            data2 = target.data.copy()
+            
+            # pad columns if different shapes
+            sizes_1 = [x.shape[1] for x in [data1, data2]]
+            C = max(sizes_1)
+            y = data1[:, 0:C]
+            missing = C - y.shape[1]
+            add = np.zeros((y.shape[0], missing))
+            data1 = np.append(y, add, axis=1)
+        else:
+            data2 = target.copy()
 
         if axis == 1:
             data1 = data1.T
             data2 = data2.T
 
-        out = dict()
+        out = {}
         if method in ['deterministic_srm', 'probabilistic_srm']:
-            if n_features is None:
-                n_features = data1.shape[0]
-            if method == 'deterministic_srm':
-                srm = DetSRM(features=n_features, *args, **kwargs)
-            elif method == 'probabilistic_srm':
-                srm = SRM(features=n_features, *args, **kwargs)
-            srm.fit([data1, data2])
-            source.data = srm.transform([data1, data2])[0].T
-            common.data = srm.s_.T
-            out['transformed'] = source
-            out['common_model'] = common
-            out['transformation_matrix'] = srm.w_[0]
+            if data2.shape[0] != data1.shape[0]:
+                raise ValueError("The number of timepoints(TRs) does not match the model.")
+
+            A = data1.T.dot(data2)
+
+            # # Solve the Procrustes problem
+            U, _, V = np.linalg.svd(A, full_matrices=False)
+
+            out['transformation_matrix'] = source
+            out['transformation_matrix'].data = U.dot(V)
+
+            out['transformed'] = data1.dot(out['transformation_matrix'].data)
+            out['common_model'] = target
```

Comment:
Also the common model is being set to the target? I'm very confused it looks like this method is kind of messed up right now, maybe roll back to a previous commit?

Reply:
no, this is correct. when you align a new subject to a common model, the output stats dictionary includes a common model. This is to keep the dictionary consistent with the real `stats.align` function. When you use the brain_data version you are aligning a single subject into a common space. The common space stays the same when it is output.

This is one of the things that I don't think I was doing correctly originally that I've now fixed, same with SRM. Brainiak added this method after we wrote ours. It wasn't in their original class.

Reply:
Way so if you're aligning just 2 brains won't the 'common model' be the second brain, i.e. the target and not a learned model?

Reply:
yeah exactly, which is why I'm no longer fitting a new SRM. now I just do a procrustes w/ SVD and rotate the new subject into the common model for the SRM.  The hyperalignment version is a little different because of scaling issues.